packages/playwright-core/src/server/chromium/chromium.ts

```diff
@@ -217,7 +218,19 @@ export class Chromium extends BrowserType {
       const capabilities = value.capabilities;
       let endpointURL: URL;
 
-      if (capabilities['se:cdp']) {
+      if (process.env.SELENIUM_WS_ENDPOINT) {
+        // support selenium successors (like selenoid)
+        progress.log(`<selenium> using some selen-project`);
+        let wsEndpoint = process.env.SELENIUM_WS_ENDPOINT;
+
+        if (SESSION_ID_PLACEHOLDER.test(wsEndpoint))
```

Comment:
Placeholders implementation is not needed by selenoid itself, but can be used by another selen-project in which `sessionId` might be in the middle of the url (like for selenium4 - `ws://localhost:4444/session/{{SESSION_ID}}/se/cdp`).